### ISSUE TITLE
Commit button is not enabled in the UI

### ISSUE THREAD

**Issue by sjothi** [mohamicorp/editor-for-stash-issues#29]
Hi,
Consider this case,
Add 2 branch permissions:
Pattern \* to limit write access to user1
Pattern feature/\* to limit write access to user2

According to the above condition user2 should have write access to "feature/TEST-1" branch. But when i select "feature/TEST-1" from the dropdown and if i click edit on a particular file the commit button is not enabled. I am evaluation license. 
This is very critical bug i guess. Let me know if i am missing something.

**Comment by mdavoodi** [comment]
Are you able to push and pull to that branch as user2? The editor branch permissions just uses Bitbucket's underlying permissions, and I believe in this scenario user1's permissions trumps user2s. I can confirm this is the default behavior of bitbucket in a few hours.

**Comment by sjothi** [comment]
1. Yes. I am able to push and pull as user2 from command line. But only on editor plugin the Commit button is disabled.

Can you please let me know what is the issue ? Take this as urgent request. Because we are planning to procure a license for 2000 Users and if this is an issue i have to talk to my management.

**Comment by mdavoodi** [comment]
It looks like this bug only happens when you have not committed anything to the branch yet. Atlassian's RefRestrictionService wont resolve the commit to the correct branch if it has the initial hash for an empty repository. I was able to fix this by using a different call in the refRestrictionSerivce. We currently have a fix for it implemented. Once the new plugin has finished going through our integration tests and uploaded to the marketplace, I'll comment here again so you can try it out.

**Comment by mdavoodi** [comment]
We've gone ahead and released a fix, could you confirm it fixed the issue you were seeing? https://marketplace.atlassian.com/plugins/com.mohamicorp.stash.plugin.editor-for-stash/versions

**Comment by sjothi** [comment]
Hi,
Thanks for fixing this so soon. But it looks like bug fix version has been released only for Bitbucket. I am using stash-3.11.4. Can we have bug fix version for this as well ?

**Comment by mdavoodi** [comment]
We're having our developers take a look at it. The API for permissions is different in Stash, so we can't apply the same fix. I'll post again once we have it sorted out.

**Comment by sjothi** [comment]
May i know the ETA to get fix version for Stash-3.11.4 ? ;)

**Comment by mdavoodi** [comment]
A day or two tops.

**Comment by sjothi** [comment]
Ok. Cool.

**Comment by sjothi** [comment]
Please let me know if you have any updates.

**Comment by mdavoodi** [comment]
We just pushed a bugfix for Stash. Let me know if the 1.5.3 version we pushed fixes the errors you were seeing.

It looks like the bug was caused by the branch permission restructuring that happened in Stash 3.10. The old branch permission API doesn't seem to work with the new branch permission scheme they added. We switched to using refRestrictionService, which was added in 3.10 and works properly.

We're only going to do be doing bug fixes for the Stash branch of the plugin, and won't be adding new features. I'd recommend upgrading to Bitbucket 4.0 if you want to enjoy the new features we're planning (like adding files).
